core/src/test/java/nl/nn/adapterframework/pipes/SkipPipeTest.java

```diff
@@ -0,0 +1,54 @@
+package nl.nn.adapterframework.pipes;
+
+
+import nl.nn.adapterframework.core.PipeRunResult;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+
+/**
+ * SkipPipe Tester.
+ *
+ * @author <Sina Sen>
+ */
+public class SkipPipeTest extends PipeTestBase<SkipPipe>{
+
+
+
+    @Override
+    public SkipPipe createPipe() {
+        return new SkipPipe();
+    }
+
+    /**
+     * Method: setSkip(int skip)
+     */
+    @Test
+    public void testDoPipeSkip3Read2WithString() throws Exception {
+        pipe.setSkip(3); pipe.setLength(2);
+        PipeRunResult res = doPipe(pipe, "0123456", session);
+        assertEquals(res.getResult().asString(), "34");
```

Comment:
Try to not test too many things in a single junittest
It will be harder to find issues if something might break in the future.
Typically in a junit test you test the smallest possible bit of code, making sure that the entire code as a whole keeps performing well.

Reply:
disected the test into smaller tests